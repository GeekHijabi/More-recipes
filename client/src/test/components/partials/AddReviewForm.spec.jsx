import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedReviewForm, { AddReviewForm }
  from '../../../js/components/Partials/AddReviewForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const review = {
  review: 'nice'
};

const props = {
  apiRecipeReview: jest.fn(() => Promise.resolve({ ok: true })),
  recipeId: 1
};
describe('Component: AddReviewForm', () => {
  beforeEach(() => {
    global.toastr = {
      success: () => {},
      error: () => {}
    };
  });

  describe('AddReviewForm in snapshot', () => {
    it('should render correctly', () => {
      const forgotPassword = shallow(<AddReviewForm {...props} />);
      expect(forgotPassword).toMatchSnapshot();
    });
  });

  describe('onChange', () => {
    it('should change event', () => {
      const event = {
        target: { name: 'review', value: 'nice recipe' },
      };
      const forgotPassword = shallow(<AddReviewForm {...props} />);
      forgotPassword.instance().onChange(event);
      expect(forgotPassword.instance().state.review).toEqual('nice recipe');
    });
  });

  describe('onClick', () => {
    it('should submit form', () => {
      const event = {
        preventDefault: jest.fn()
      };
      const submit = shallow(<AddReviewForm {...props} />);
      submit.setState(review);
      submit.setState({ hasError: true, errorMessage: 'error', errors: {} });
      submit.instance().onClick(event);
    });
  });

  describe('isValid', () => {
    it('should check if data is valid', () => {
      const errors = {
        email: ''
      };
      const event = {
        preventDefault: jest.fn()
      };
      const isNotvalid = shallow(<AddReviewForm {...props} />);
      // isNotvalid.setState(userInput);
      isNotvalid.setState(errors);
      isNotvalid.instance().isValid(event);
    });
  });
  describe('Connected RecipeAdmin component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        recipe: {
          recipe: {
            Reviews: []
          },
        }
      });
      const wrapper = shallow(<ConnectedReviewForm store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});

