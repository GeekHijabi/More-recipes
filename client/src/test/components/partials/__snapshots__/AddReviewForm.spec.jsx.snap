// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AddReviewForm AddReviewForm in snapshot should render correctly 1`] = `
<div
  className="review-form"
  placeholder="Add review"
>
  <div
    className="input-control add-review"
  >
    <label
      htmlFor="review"
      id="label"
      style={
        Object {
          "margin": "0 auto",
          "width": "100%",
        }
      }
    >
      <textarea
        className="review-edit"
        col="3"
        id="add-review"
        name="review"
        onChange={[Function]}
        placeholder="input your review"
        type="text"
        value=""
      />
    </label>
  </div>
  <button
    className="review-btn"
    onClick={[Function]}
    type="button"
  >
    Add review
  </button>
</div>
`;
